pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1079,30 +1079,16 @@ func (og *operationGenerator) GenerateUnmapDeviceFunc(
 			return deviceToDetach.GenerateError("UnmapDevice failed", err)
 		}
 
-		// Execute tear down device
-		unmapErr := blockVolumeUnmapper.TearDownDevice(globalMapPath, deviceToDetach.DevicePath)
-		if unmapErr != nil {
-			// On failure, return error. Caller will log and retry.
-			return deviceToDetach.GenerateError("UnmapDevice.TearDownDevice failed", unmapErr)
-		}
-
-		// Plugin finished TearDownDevice(). Now globalMapPath dir and plugin's stored data
-		// on the dir are unnecessary, clean up it.
-		removeMapPathErr := og.blkUtil.RemoveMapPath(globalMapPath)
-		if removeMapPathErr != nil {
-			// On failure, return error. Caller will log and retry.
-			return deviceToDetach.GenerateError("UnmapDevice failed", removeMapPathErr)
-		}
-
 		// The block volume is not referenced from Pods. Release file descriptor lock.
 		glog.V(4).Infof("UnmapDevice: deviceToDetach.DevicePath: %v", deviceToDetach.DevicePath)
 		loopPath, err := og.blkUtil.GetLoopDevice(deviceToDetach.DevicePath)
 		if err != nil {
 			if err.Error() == volumepathhandler.ErrDeviceNotFound {
-				glog.Warningf(deviceToDetach.GenerateMsgDetailed("UnmapDevice: Couldn't find loopback device which takes file descriptor lock", fmt.Sprintf("device path: %q", deviceToDetach.DevicePath)))
+				glog.Warningf(deviceToDetach.GenerateMsgDetailed("UnmapDevice: Couldn't find loopback device which takes file descriptor lock",
+					fmt.Sprintf("device path: %q", deviceToDetach.DevicePath)))
 			} else {
-				errInfo := "UnmapDevice.GetLoopDevice failed to get loopback device, " + fmt.Sprintf("device path: %q", deviceToDetach.DevicePath)
-				return deviceToDetach.GenerateError(errInfo, err)
+				glog.Warningf(deviceToDetach.GenerateMsgDetailed("UnmapDevice: GetLoopDevice failed to get loopback device",
```

Comment:
what other errors would make it ok to skip the error and continue?

Reply:
@msau42

Thank you for pointing it out.

I thought that I hit an issue by not skiping the error from ```GetLoopDevice``` other than ```ErrDeviceNotFound``` error, while I tested. However, I can't reproduce the issue, now.

```unmapDeviceFunc``` that is generated in ```GenerateUnmapDeviceFunc``` could be called multiple times, so there will be cases that ```GetLoopDevice``` is called after ```RemoveLoopDevice``` or ```TearDownDevice``` from previous ```unmapDeviceFunc``` calls. 
As you pointed out, in most cases, ```GetLoopDevice``` returns ```ErrDeviceNotFound``` error, so the only cases that I can think of are:
1. ```os.Stat({path})``` fails with error other than not exist error,
2. ```losetup -f {path}``` command fails with error.

I guess that the second one would much likely to happen because man page of losetup is saying that it could return erorr, but I can't make losetup return these errors.
```
RETURN VALUE
       losetup returns 0 on success, nonzero on failure.  When losetup displays the status of a loop device, it returns 1 if the device is not con‐
       figured and 2 if an error occurred which prevented determining the status of the device.
```

So, I started to think that it would be better not changing here, until we find a good reproducer. I will revert the change for skipping error.